ouroboros-consensus-cardano/test/Test/ThreadNet/Cardano.hs

```diff
@@ -203,13 +237,58 @@ prop_simple_cardano_convergence TestSetup
     epochSizeShelley :: EpochSize
     epochSizeShelley = sgEpochLength genesisShelley
 
+    -- the protocol version of the Byron era proposal
+    propPV :: CC.Update.ProtocolVersion
+    propPV =
+      if setupHardFork
+      then CC.Update.ProtocolVersion shelleyMajorVersion 0 0
+      else CC.Update.ProtocolVersion
+             byronMajorVersion
+             (byronInitialMinorVersion + 1)
```

Comment:
I don't understand this, can you explain?

Reply:
I've added a couple comments. Also see the comment on `TestSetup` and `setupHardFork`.

Does that help?

Reply:
Was meant as a response: https://github.com/input-output-hk/ouroboros-network/pull/2282#discussion_r442243653